### PR TITLE
feat(experimental): `Textfield` add new component

### PR DESCRIPTION
**Pull request by waterplea**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Build or CI related changes
- [ ] Tests related changes
- [ ] Documentation content changes

## What is the current behaviour?

Closes #

### CODE REVIEW

File: projects/experimental/components/textfield/textfield.template.html [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    <button
+        *ngIf="options.cleaner"
+        appearance="icon"
+        iconLeft="tuiIconClose"
+        size="xs"
+        tabindex="-1"
+        tuiIconButton
+        class="t-clear"
+        (click)="input!.nativeElement.value = ''"
+        (pointerdown.silent.prevent)="input!.nativeElement.focus()"
```
Comment (splincode) [inline projects/experimental/components/textfield/textfield.template.html:19]: can we override the button close template by PolymorpheusComponent?

```html
<ng-container *ngIf="options.cleaner">
    <ng-container *polymorpheusOutlet="buttonComponent as text; context: {$implicit: input}">
        {{ text }}
    </ng-container>
</ng-container>
```

**Review by splincode — APPROVED** [review]

File: projects/experimental/components/textfield/textfield.template.html (hunk 1 above)
Comment (waterplea) [reply projects/experimental/components/textfield/textfield.template.html:19]: Let's not overcomplicate things. You can always disable built-in cleaner and add your own inside content.

File: projects/demo/src/modules/experimental/textfield/examples/1/index.html [hunk 2]
```diff
@@ -0,0 +1,219 @@
+<div
+    class="new"
```
Comment (vladimirpotekhin) [inline projects/demo/src/modules/experimental/textfield/examples/1/index.html:2]: unused class ?

File: projects/demo/src/modules/experimental/textfield/examples/1/index.html [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+        <tui-icon
+            icon="tuiIconBellLarge"
+            ngProjectAs="tuiSide"
+        ></tui-icon>
+        <tui-tooltip content="I am a hint"></tui-tooltip>
+    </tui-textfield>
+</div>
+
+<div
+    class="old"
```
Comment (vladimirpotekhin) [inline projects/demo/src/modules/experimental/textfield/examples/1/index.html:132]: same

File: projects/experimental/components/label/label.directive.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+import {TuiLabelComponent} from './label.component';
+
+@Directive({
+    standalone: true,
+    selector: 'label[tuiLabel]',
+})
+export class TuiLabelDirective {
+    // @ts-ignore
+    private readonly nothing = tuiWithStyles(TuiLabelComponent);
```
Comment (vladimirpotekhin) [inline projects/experimental/components/label/label.directive.ts:13]: `nothing` is expected name?)

File: projects/demo/src/modules/experimental/textfield/examples/1/index.html [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+    >
+        <label tuiLabel>I am a label</label>
+        <input
+            placeholder="I am placeholder"
+            tuiTextfieldd
+            [(ngModel)]="value"
+        />
+        <tui-icon
+            icon="tuiIconBellLarge"
+            ngProjectAs="tuiSide"
```
Comment (vladimirpotekhin) [inline projects/demo/src/modules/experimental/textfield/examples/1/index.html:21]: Is it necessary? I didn't find `select="tuiSide"`

File: projects/experimental/components/textfield/textfield.template.html [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+<ng-content select="label"></ng-content>
+<span
+    class="t-content"
+    (waResizeObserver)="side = $event[0].contentRect.width"
+>
+    <ng-content></ng-content>
+    <button
+        *ngIf="options.cleaner"
+        appearance="icon"
+        iconLeft="tuiIconClose"
```
Comment (vladimirpotekhin) [inline projects/experimental/components/textfield/textfield.template.html:13]: we need to somehow override icon in proprietary theme

File: projects/experimental/components/label/label.directive.ts (hunk 4 above)
Comment (waterplea) [reply projects/experimental/components/label/label.directive.ts:13]: Well, we can add constructor and run that code inside of it, as an alternative. I was also thinking of calling it void. There's no other way to run initialization code, unfortunately.

File: projects/demo/src/modules/experimental/textfield/examples/1/index.html (hunk 5 above)
Comment (waterplea) [reply projects/demo/src/modules/experimental/textfield/examples/1/index.html:21]: Old idea

File: projects/experimental/components/textfield/textfield.template.html (hunk 6 above)
Comment (waterplea) [reply projects/experimental/components/textfield/textfield.template.html:13]: Sure, I'll add this later on.

**Review by vladimirpotekhin — APPROVED** [review]

**Review by nsbarsukov — APPROVED** [review]

**Pull request merged**